src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java

```diff
@@ -66,9 +73,24 @@
      */
     public static final String MSG_KEY = "import.illegal";
 
+    /** The compiled regular expressions for packages. */
+    private final List<Pattern> illegalPkgsRegexps = new ArrayList<>();
+
+    /** The compiled regular expressions for classes. */
+    private final List<Pattern> illegalClassesRegexps = new ArrayList<>();
+
     /** List of illegal packages. */
     private String[] illegalPkgs;
 
+    /** List of illegal classes. */
+    private List<String> illegalClasses = new ArrayList<>();
```

Comment:
Why not make this an array like `illegalPkgs`? It is used in the exact same way.